### PR TITLE
Fixing id encoding issues when using special characters for customers already on ComputeGateway

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
PR #29944 - https://github.com/Azure/azure-sdk-for-java/pull/29944 changed the encoding of special characters in the documentId to make sure it works with RoutingGateway - it turned out that ComputeGateway and RoutingGateway expect different encoding (RoutingGateway expects id in the Uri to be double-percent-encoded - while Compute expects it to be only single percent-encoded. Aftersome discussion we decided that ComputeGateway behavior is actually correct - and making sure SDK works weel even when id has special characters in ComputeGateway means there are no regressions for customers on RoutingGateway (same failures as before #29944) - but at least it allows us to unblock customers who want to use special characters in id by moving them from RoutingGateway to ComputeGateway. The automatic migration from RoutingGateway to ComputeGateway will also accelerate - and over time the issues in RoutingGateway will vanish.

This PR moves back to a behavior working well with ComuteGateway - and ass tests against RountingGateway and ComputeGateway in the emulator.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosAsyncClientTest.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public final void setTestName(Method method) {
         if (this.clientBuilder.getConnectionPolicy() != null && this.clientBuilder.configs() != null) {
             String connectionMode = this.clientBuilder.getConnectionPolicy().getConnectionMode() == ConnectionMode.DIRECT
                     ? "Direct " + this.clientBuilder.configs().getProtocol()
-                    : "Gateway";
+                    : this.clientBuilder.getEndpoint().contains(":8903") ? "ComputeGW" : "Gateway";
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosAsyncClientTest.java:49]: Can we have this port number in a constant?

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java [hunk 2]
```diff
@@ -463,7 +578,8 @@ public void idWithLineFeed() {
     private void executeTestCase(TestScenario scenario) {
         TestScenarioExpectations expected =
             this.getConnectionPolicy().getConnectionMode() == ConnectionMode.DIRECT ?
-                scenario.Direct : scenario.Gateway;
+                scenario.Direct : this.getClientBuilder().getEndpoint().contains(":8903") ?
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java:581]: same here - port number constant.

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosAsyncClientTest.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosAsyncClientTest.java:49]: Resolved in next iteration.

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java:581]: Resolved in next iteration

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java [hunk 3]
```diff
@@ -616,6 +743,8 @@ public TestScenario(
 
         public TestScenarioExpectations Gateway;
 
+        public TestScenarioExpectations ComputeGateway;
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java:746]: nit: variable name starts with lowercase

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java (hunk 3 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosItemIdEncodingTest.java:746]: Fixed in next iteration.

**Pull request merged**